stackit/internal/services/iaas/network/resource.go

```diff
@@ -450,59 +632,149 @@ func toCreatePayload(ctx context.Context, model *Model) (*iaas.CreateNetworkPayl
 	if model == nil {
 		return nil, fmt.Errorf("nil model")
 	}
+	addressFamily := &iaas.CreateNetworkAddressFamily{}
+
+	modelIPv6Nameservers := []string{}
+	for _, ipv6ns := range model.IPv6Nameservers.Elements() {
+		ipv6NameserverString, ok := ipv6ns.(types.String)
+		if !ok {
+			return nil, fmt.Errorf("type assertion failed")
+		}
+		modelIPv6Nameservers = append(modelIPv6Nameservers, ipv6NameserverString.ValueString())
+	}
+
+	if !(model.IPv6Prefix.IsNull() || model.IPv6PrefixLength.IsNull() || model.IPv6Nameservers.IsNull()) {
+		addressFamily.Ipv6 = &iaas.CreateNetworkIPv6Body{
+			Nameservers:  &modelIPv6Nameservers,
+			Prefix:       conversion.StringValueToPointer(model.IPv6Prefix),
+			PrefixLength: conversion.Int64ValueToPointer(model.IPv6PrefixLength),
+		}
+
+		if model.NoIPv6Gateway.ValueBool() {
+			addressFamily.Ipv6.Gateway = iaas.NewNullableString(nil)
+		} else if !(model.IPv6Gateway.IsUnknown() || model.IPv6Gateway.IsNull()) {
+			addressFamily.Ipv6.Gateway = iaas.NewNullableString(conversion.StringValueToPointer(model.IPv6Gateway))
+		}
+	}
 
-	modelNameservers := []string{}
-	for _, ns := range model.Nameservers.Elements() {
-		nameserverString, ok := ns.(types.String)
+	modelIPv4Nameservers := []string{}
+	var modelIPv4List []attr.Value
+
+	if !(model.IPv4Nameservers.IsNull() || model.IPv4Nameservers.IsUnknown()) {
+		modelIPv4List = model.IPv4Nameservers.Elements()
+	} else {
+		modelIPv4List = model.Nameservers.Elements()
+	}
+
+	for _, ipv4ns := range modelIPv4List {
+		ipv4NameserverString, ok := ipv4ns.(types.String)
 		if !ok {
 			return nil, fmt.Errorf("type assertion failed")
 		}
-		modelNameservers = append(modelNameservers, nameserverString.ValueString())
+		modelIPv4Nameservers = append(modelIPv4Nameservers, ipv4NameserverString.ValueString())
+	}
+
+	if !model.IPv4Prefix.IsNull() || !model.IPv4PrefixLength.IsNull() || !model.IPv4Nameservers.IsNull() || !model.Nameservers.IsNull() {
+		addressFamily.Ipv4 = &iaas.CreateNetworkIPv4Body{
+			Nameservers:  &modelIPv4Nameservers,
+			Prefix:       conversion.StringValueToPointer(model.IPv4Prefix),
+			PrefixLength: conversion.Int64ValueToPointer(model.IPv4PrefixLength),
+		}
+	}
+
+	if model.NoIPv4Gateway.ValueBool() {
+		addressFamily.Ipv4.Gateway = iaas.NewNullableString(nil)
+	} else if !(model.IPv4Gateway.IsUnknown() || model.IPv4Gateway.IsNull()) {
+		addressFamily.Ipv4.Gateway = iaas.NewNullableString(conversion.StringValueToPointer(model.IPv4Gateway))
 	}
```

Comment:
Won't this lead to a null pointer exception if the `if` above (line 677) is not entered? Because the `if` does not run, `addressFamily.Ipv4` will be null here, right?

Reply:
For IPv6 this if inside the other `if`

Reply:
Same for the update method

Reply:
It never caused a nil pointer exception during tests. 

But it might make sense to move in the above if block, cause gateway can only be set with prefix set when creating a network.